pkg/cri/server/sandbox_portforward_windows.go

```diff
@@ -17,61 +17,101 @@
 package server
 
 import (
-	"bytes"
 	"context"
 	"fmt"
 	"io"
+	"net"
+	"time"
 
-	"k8s.io/utils/exec"
-
-	sandboxstore "github.com/containerd/containerd/v2/pkg/cri/store/sandbox"
-	cioutil "github.com/containerd/containerd/v2/pkg/ioutil"
+	"github.com/containerd/log"
+	netutils "k8s.io/utils/net"
 )
 
 func (c *criService) portForward(ctx context.Context, id string, port int32, stream io.ReadWriter) error {
-	stdout := cioutil.NewNopWriteCloser(stream)
-	stderrBuffer := new(bytes.Buffer)
-	stderr := cioutil.NewNopWriteCloser(stderrBuffer)
-	// localhost is resolved to 127.0.0.1 in ipv4, and ::1 in ipv6.
-	// Explicitly using ipv4 IP address in here to avoid flakiness.
-	cmd := []string{"wincat.exe", "127.0.0.1", fmt.Sprint(port)}
-	err := c.execInSandbox(ctx, id, cmd, stream, stdout, stderr)
+	sandbox, err := c.sandboxStore.Get(id)
 	if err != nil {
-		return fmt.Errorf("failed to execute port forward in sandbox: %s: %w", stderrBuffer.String(), err)
+		return fmt.Errorf("failed to find sandbox %q in store: %w", id, err)
 	}
-	return nil
-}
 
-func (c *criService) execInSandbox(ctx context.Context, sandboxID string, cmd []string, stdin io.Reader, stdout, stderr io.WriteCloser) error {
-	// Get sandbox from our sandbox store.
-	sb, err := c.sandboxStore.Get(sandboxID)
-	if err != nil {
-		return fmt.Errorf("failed to find sandbox %q in store: %w", sandboxID, err)
+	var podIP string
+	if !hostNetwork(sandbox.Config) {
+		// get ip address of the sandbox
+		podIP, _, err = c.getIPs(sandbox)
```

Comment:
In a dual stack setup, is the IPv4 address always the podIP and then the additional addresses is the IPv6 address? Curious about that.

Reply:
That's what the assumption on linux side seems to be. Having said that, not too familiar with the networking aspects of containerd. If this needs to be smth else, could you suggest? You might have more experience on the networking side.

Reply:
I will need to look into the kubelet side here. I can't remember if we change PodIP to always be ipv4 or is it driven by however the CNI returns them. Aka if the CNI result returns ipv6 address first then its PodIP. I will circle back.

Reply:
Sorry I thought this happened in the kubelet for some reason. 
https://github.com/containerd/containerd/blob/93573f5cf3054fd7505e738fef3887532dde1530/pkg/cri/server/sandbox_run.go#L592

Reply:
Hopefully I am not turning this into an unbounded refactor. We can always capture todos and tackle later :-)

Reply:
not sure I understand the concern. The link you are pointing to is how the sandbox IPs are even set. I am only trying to read them from the windows port forwarding.